### PR TITLE
Add Creeper Mock

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Add Creeper Mock

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			this.powered = value;
+		}
+	}
+
+	private boolean callPowerEvent(CreeperPowerEvent.PowerCause cause)
+	{
+		CreeperPowerEvent event = new CreeperPowerEvent( this, cause);
+		this.getServer().getPluginManager().callEvent(event);
+		return event.isCancelled();
+	}
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]:
```suggestion
		if (!new CreeperPowerEvent(this, cause).callEvent())
		{
			this.powered = value;
		}
	}
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	@Override
+	public void ignite()
+	{
+		setIgnited(true);
+	}
+
+	@Override
+	public void setIgnited(boolean ignited)
+	{
+		if (isIgnited() != ignited)
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]: This can be inverted

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java (hunk 1 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]: This is how it's handled in CraftBukkit, thats why I think we should keep it

Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]: It's only used in one place. We don't need to copy CraftBukkit's code 1 for 1, just as long as the functionality is the same. If it needs to be used in more places in the future, then a method would be good. But for now, doing it this way is much cleaner.

File: src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	{
+		return this.powered;
+	}
+
+	@Override
+	public void setPowered(boolean value)
+	{
+		CreeperPowerEvent.PowerCause cause = powered ? CreeperPowerEvent.PowerCause.SET_ON : CreeperPowerEvent.PowerCause.SET_OFF;
+
+		if (new CreeperPowerEvent( this, cause).callEvent())
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]:
```suggestion
		if (new CreeperPowerEvent(this, cause).callEvent())

File: src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		{
+			this.powered = value;
+		}
+	}
+
+	@Override
+	public void setMaxFuseTicks(int ticks)
+	{
+		Preconditions.checkArgument(ticks >= 0, "Ticks need to be bigger than 0");
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/CreeperMock.java]:
```suggestion

File: src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java [hunk 5]
```diff
@@ -1235,6 +1237,14 @@ void testSpawnBee()
 		assertTrue(entity.isValid());
 	}
 
+	@Test
+	void testSpawnCreeper()
+	{
+		WorldMock world = new WorldMock(Material.DIRT, 3);
+		Entity entity = world.spawnEntity(new Location(world, 0, 0, 0), EntityType.CREEPER);
+		assertInstanceOf(CreeperMock.class, entity);
+    }
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java]:
```suggestion
	}

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM

**Pull request merged**